src/core/etl/src/Flow/ETL/Function/ArrayKeys.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Flow\ETL\Function;
+
+use Flow\ETL\Row;
+
+final class ArrayKeys extends ScalarFunctionChain
+{
+    public function __construct(private readonly ScalarFunction|array $array)
+    {
+    }
+
+    public function eval(Row $row) : mixed
+    {
+        $array = (new Parameter($this->array))->asArray($row);
+
+        if (!\is_array($array)) {
```

Comment:
```suggestion
        if (null === $array) {
```

Reply:
does it give anything?

Reply:
Strict null check will be cheaper, and `asArray()` can return either array or null.

Reply:
I did a quick benchmark on 10mln iterations and the differences are almost invisible, looks like microoptimisation in one place won't generate enough impact.
It might be worth using rector/phpstan to enforce that everywhere